### PR TITLE
Add CI check for DO_NOT_SUBMIT comments in go files.

### PR DESCRIPTION
**Pull request by MitchAman**
# Description

Add a CI check which fails when `// DO_NOT_SUBMIT` is found in *.go files that were modified in a PR.

Example failure: https://github.com/solo-io/gloo/runs/3654089442?check_suite_focus=true

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: changelog/v1.9.0-beta19/do-not-submit.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+changelog:
+  - type: NON_USER_FACING
+    description: Add a CI check which fails when `// DO_NOT_SUBMIT` is found in *.go files that were modified in a PR.
```
Comment (kdorosh) [inline changelog/v1.9.0-beta19/do-not-submit.yaml:3]: what files do we generate have this header today?

Comment (MitchAman) [reply changelog/v1.9.0-beta19/do-not-submit.yaml:3]: What do you mean by that? No files contain "DO_NOT_SUBMIT" at the moment (besides those introduced in this PR, but they aren't go files

**Review by sam-heilbron — COMMENTED** [review]
I believe the goal of this is to avoid merging PRs until they are 'ready'. How does this GHA differ from the behavior of the 'work in progress' label or creating a PR as a draft?

**Comment by MitchAman** [comment]
> I believe the goal of this is to avoid merging PRs until they are 'ready'. How does this GHA differ from the behavior of the 'work in progress' label or creating a PR as a draft?

The behavior is largely the same as the 'work in progress' label, though `// DO_NOT_SUBMIT` can be added before a PR is opened while it's still fresh in your mind, rather than having to remember to apply the 'work in progress' label.

One scenario (although not actually supported yet because the file extension isn't *.go) where this would be useful is within a gloo PR which depends on a solo-kit PR that's still in review.  Imagine that within gloo's `go.mod` the solo-kit version is set to the latest commit of the solo-kit PR. We don't want the gloo PR to get merged until the solo-kit PR is merged and released, so that we can update the specific commit reference to the newly released version.  Dropping the following comment in `go.mod`:

1. makes it clear what still needs to be done in the PR
2. can be applied immediately when changing the code (rather than after PR is opened)
3. provides a layer of protection if one forgets to apply 'work in progress' and/or a reviewer doesn't catch it
```
// DO_NOT_SUBMIT: Update to new solo-kit version once it's released
github.com/solo-io/solo-kit v0.22.1-0.20210920203028-060e333b9a2d
```
In this scenario, we want the gloo PR to be reviewed - everything is ready to go, it's just blocked by the solo-kit change, so a draft PR feels inappropriate. The 'work in progress' label could be used to protect against merging in this scenario, but one needs to remember to apply it. The 'work in progress' label also does not provide any context around what part of the PR is still being worked on, while the comment makes it very clear.

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**